modules/core/src/smithy4s/Document.scala

```diff
@@ -35,6 +35,17 @@ sealed trait Document extends Product with Serializable {
 
   override def toString(): String = this.show
 
+  def name: String = this match {
+    case DNumber(_)  => "number"
```

Comment:
**suggestion:** these are lowercase, but AFAICS the usage of `DocumentDecoder.instance` uses uppercase names for the expected JSON shape. Would you like to update the call sites so that it's consistenly lowercase? (or the opposite)

<img width="757" alt="image" src="https://github.com/disneystreaming/smithy4s/assets/894884/478077ca-39e5-4887-8a69-2e51e79b4bef">